packages/query-core/src/utils.ts

```diff
@@ -330,3 +330,31 @@ export function keepPreviousData<T>(
 ): T | undefined {
   return previousData
 }
+
+export class LimitedLengthArray<T> {
+  private items: T[]
+  private length: number | undefined
```

Comment:
in v5, we're using private class fields please:

```
#items: T[],
#length: number | undefined
```

But I'm wondering if a class is the right abstraction / worth the overhead. Wouldn't two immutable util functions achieve the same goal? Something like:

```
const addToEnd = (items, item, max = 0) => {
  const new = [...items, item]
  return max > new.length ? new.slice(1) : new
}

const addToStart = (items, item, max = 0) => {
  const new = [item, ...items]
  return max > new.length ? new.slice(0, -1) : new
}
```

Reply:
Ok let me refactor that with util functions.

Reply:
Done in 434af38abd73ebc4a38a72ca87fbca8f359dbb7a